python/lsst/cell_coadds/_fits.py

```diff
@@ -75,11 +110,61 @@ def __init__(self, filename: str) -> None:
 
         self.filename = filename
 
+    @classmethod
+    def isCompatibleWith(cls, written_version: str, /) -> bool:
+        """Check if the serialization version is compatible with the reader.
+
+        Notes
+        -----
+        This accepts the other version as a positional argument only.
+
+        Parameters
+        ----------
+        written_version: `str`
+            The version to compare against.
+
+        Returns
+        -------
+        compatible : `bool`
+            Whether the reader is compatible with ``written_version``.
+        """
+        this_version = version.parse(FILE_FORMAT_VERSION)
+        other_version = version.parse(written_version)
+
+        if this_version.major != other_version.major:
+            return False
+
+        if this_version.minor < other_version.minor:
+            return False
```

Comment:
This feels too pessimistic.  It's shouldn't be that hard for us to ignore header keys, table columns, and HDUs we don't recognize.